tokio/src/runtime/handle.rs

```diff
@@ -202,7 +202,105 @@ impl Handle {
         handle
     }
 
-    /// TODO: write docs if this is a good direction
+    /// Run a future to completion on this `Handle`'s associated `Runtime`.
+    ///
+    /// This runs the given future on the runtime, blocking until it is
+    /// complete, and yielding its resolved result. Any tasks or timers which
+    /// the future spawns internally will be executed on the runtime.
+    ///
+    /// The behavior for multi threaded vs current thread schedulers is the same
+    /// as [`Runtime::block_on`]. See the docs of [`Runtime::block_on`] for more
+    /// details.
+    ///
+    /// # If the runtime has been shutdown
+    ///
+    /// If the `Handle`'s associated `Runtime` has been shutdown, through
+    /// [`Runtime::shutdown_background`] or [`Runtime::shutdown_timeout`], and
+    /// `Handle::block_on` is used it might return an error or panic. The exact
+    /// behavior depends on the types of futures used.
```

Comment:
A runtime can also be shut down by dropping it.

Reply:
Good point! Added note about that.